src/matter/interaction/InteractionClient.ts

```diff
@@ -131,7 +131,7 @@ export class InteractionClient {
             if (result !== undefined) {
                 const resultCode = result.result.code;
                 if (resultCode !== ResultCode.Success) throw new Error(`Received non-success result: ${resultCode}`);
-                if (responseTemplate !== NoResponseT) throw new Error("A response was expected for this command");
+                if ((responseTemplate as any) !== NoResponseT) throw new Error("A response was expected for this command");
```

Comment:
Not sure why the ts compiler is now complaining here and what is its problem.
It says that there is no overlap of responseTemplate and NoResponseT but this is not true.

responseTemplate can be Template<void> for Command<?, void>

Well, this type check is not important , so casting to any to avoid the none-sense problem. This won't have any runtime impact.